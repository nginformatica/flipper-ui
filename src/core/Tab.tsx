import { Tab as MuiTab } from '@material-ui/core'
import React, { Component } from 'react'
import { DefaultProps } from './types'

interface TabProps extends DefaultProps {
    disabled?: boolean
    icon?: string | JSX.Element
    label?: string
    value?: unknown
    disableRipple?: boolean
}

class Tab extends Component<TabProps> {
    public static defaultProps = {
        disabled: false,
        margin: '0 4px'
    }

    public render() {
        const { style, margin, disableRipple, padding, ...otherProps } =
            this.props

        return (
            <MuiTab
                disableRipple
                style={{ margin, padding, ...style }}
                {...otherProps}
            />
        )
    }
}

export default Tab
