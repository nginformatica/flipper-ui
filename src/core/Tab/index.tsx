import { Tab as MuiTab } from '@material-ui/core'
import React from 'react'
import { DefaultProps } from '../types'

interface TabProps extends DefaultProps {
    disabled?: boolean
    icon?: string | JSX.Element
    label?: string
    value?: unknown
    disableRipple?: boolean
}

const Tab = ({
    style,
    margin,
    disableRipple,
    padding,
    ...otherProps
}: TabProps) => (
    <MuiTab
        disableRipple
        style={{ margin, padding, ...style }}
        {...otherProps}
    />
)

export default Tab
