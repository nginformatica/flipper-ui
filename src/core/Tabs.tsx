import { Tabs as MuiTabs } from '@material-ui/core'
import React, { ReactNode } from 'react'
import { DefaultProps } from './types'

interface TabsProps extends DefaultProps {
    centered?: boolean
    value: string | number
    color?: 'default' | 'inherit' | 'primary' | 'secondary'
    variant?: 'standard' | 'scrollable' | 'fullWidth'
    children?: ReactNode
    indicatorColor?: 'primary' | 'secondary'
    onChange?: (event: object, value: number) => void
}

interface IClasses {
    classes?: {
        default: string
        inherit: string
        primary: string
        secondary: string
        indicator: string
    }
}

const Tabs = ({
    children,
    centered = true,
    padding = '6px 0 0',
    style,
    margin,
    variant = 'standard',
    indicatorColor = 'primary',
    ...otherProps
}: TabsProps & IClasses) => {
    return (
        <MuiTabs
            indicatorColor={indicatorColor}
            centered={centered}
            {...otherProps}
            variant={variant}
            style={{ padding, margin, ...style }}>
            {children}
        </MuiTabs>
    )
}

export default Tabs
