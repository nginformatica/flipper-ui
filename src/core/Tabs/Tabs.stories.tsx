import React from 'react'
import { Meta, StoryFn } from '@storybook/react'
import Tabs from '.'
import Tab from '../Tab'

export default {
    title: 'Tabs',
    component: Tabs
} as Meta<typeof Tabs>

const Template: StoryFn<typeof Tabs> = args => <Tabs {...args} />

export const Default = Template.bind({})
Default.args = {
    value: 0,
    children: (
        <>
            <Tab label='Profile' />
            <Tab label='Enterprise' />
            <Tab label='Billing' />
        </>
    )
}
