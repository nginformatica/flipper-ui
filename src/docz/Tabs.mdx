---
name: Tabs
route: /tabs
---

import Tabs from '../core/Tabs'
import Tab from '../core/Tab'
import { Playground } from 'docz'

# Tabs

## Demo
<Playground>
    <Tabs value={ 0 }>
        <Tab label='Profile' />
        <Tab label='Enterprise' />
        <Tab label='Billing' />
    </Tabs>
</Playground>
