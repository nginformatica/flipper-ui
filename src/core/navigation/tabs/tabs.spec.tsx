import React from 'react'
import { render, screen } from '@testing-library/react'
import Tab from '@/core/navigation/tab'
import Tabs from '.'

describe('Tabs', () => {
    it('should render', () => {
        render(
            <Tabs value={0}>
                <Tab label='Profile' />
                <Tab label='Enterprise' />
            </Tabs>
        )

        const tabActive = screen.getByText('Profile').parentElement
        const tabInactive = screen.getByText('Enterprise').parentElement

        expect(tabActive?.classList).toContain('Mui-selected')
        expect(tabInactive?.classList).not.toContain('Mui-selected')
    })

    it('should render with custom styles', () => {
        const { container } = render(
            <Tabs style={{ color: 'red' }} margin={10} padding={5} value={0}>
                <Tab label='Profile' />
                <Tab label='Enterprise' />
            </Tabs>
        )

        expect(container.firstElementChild).toHaveProperty('style.color', 'red')
        expect(container.firstElementChild).toHaveProperty(
            'style.margin',
            '10px'
        )
        expect(container.firstElementChild).toHaveProperty(
            'style.padding',
            '5px'
        )
    })

    it('should match snapshot', () => {
        const { container } = render(
            <Tabs style={{ color: 'red' }} margin={10} padding={5} value={0}>
                <Tab label='Profile' />
                <Tab label='Enterprise' />
            </Tabs>
        )

        expect(container).toMatchSnapshot()
    })
})
