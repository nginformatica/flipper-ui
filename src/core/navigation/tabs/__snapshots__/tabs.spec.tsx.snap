// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should match snapshot 1`] = `
<div>
  <div
    class="MuiTabs-root"
    style="padding: 5px; margin: 10px; color: red;"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed"
      style="overflow: hidden;"
    >
      <div
        class="MuiTabs-flexContainer MuiTabs-centered"
        role="tablist"
      >
        <button
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
          role="tab"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            Profile
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            Enterprise
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        class="PrivateTabIndicator-root-9 PrivateTabIndicator-colorPrimary-10 MuiTabs-indicator"
        style="left: 0px; width: 0px;"
      />
    </div>
  </div>
</div>
`;
